test/benchmark/vqe_application.py

```diff
@@ -40,34 +33,82 @@ def __init__(self,
+        
+        driver = PySCFDriver(atom=mol_string, unit=UnitsType.ANGSTROM, basis='sto3g')
+        es_problem = ElectronicStructureProblem(driver)
+        qubit_converter = QubitConverter(JordanWignerMapper())
+        max_evals_grouped = 1024
+        
+        from qiskit.algorithms.optimizers import SLSQP
+        optimizer = SLSQP(maxiter=5000)
+        
+        from qiskit.utils import QuantumInstance
+        
+        quantum_instance = QuantumInstance(backend=simulator)
+        
+        # import logging
+        # logging.basicConfig(format='%(asctime)s %(levelname)-8s %(message)s',
+        #                     level=logging.DEBUG,
+        #                     datefmt='%Y-%m-%d %H:%M:%S')
+        
+        from qiskit_nature.algorithms import VQEUCCFactory
+        
+        vqe_solver = VQEUCCFactory(quantum_instance,
+                                   optimizer=optimizer,
+                                   include_custom=True,
+                                   max_evals_grouped=max_evals_grouped)
+        
+        from qiskit_nature.algorithms import GroundStateEigensolver
+        
+        calc = GroundStateEigensolver(qubit_converter, vqe_solver)
+        
+        res = calc.solve(es_problem)
+        
     def time_statevector(self, mol_name):
+        from qiskit.providers.aer import AerSimulator
         threads = multiprocessing.cpu_count()
         mol_string = self.mol_strings[mol_name][0]
         qubit = self.mol_strings[mol_name][1]
-        self._run_uccsd_vqe(mol_string, 'statevector', threads)
+        self._run_uccsd_vqe(mol_string, 'statevector_cpu', AerSimulator(device='CPU'), threads)
 
     def time_statevector_gpu(self, mol_name):
+        from qiskit.providers.aer import AerSimulator
         threads = 1
         mol_string = self.mol_strings[mol_name][0]
         qubit = self.mol_strings[mol_name][1]
-        self._run_uccsd_vqe(mol_string, 'statevector_gpu', threads)
+        self._run_uccsd_vqe(mol_string, 'statevector_gpu', AerSimulator(device='GPU'), threads)
+
+    def run_manual(self):
+        import timeout_decorator
+        @timeout_decorator.timeout(self.timeout)
+        def run_with_timeout (suite, method, molstring):
+            start = time()
+            eval(f'suite.time_{method}("{molstring}")')
+            return time() - start
+        
+        #for runtime in self.runtime_names:
+        for method in [ 'statevector' ]:
+            for mol_string in self.mol_strings:
+                print (f'{self.__name__},uccsd,{method},{mol_string},', end="")
+                try:
+                    elapsed = run_with_timeout(self, method, mol_string)
+                    print ('{0}'.format(elapsed))
+                except ValueError as e:
+                    print ('{0}'.format(e))
+                except:
+                    import traceback
+                    traceback.print_exc(file=sys.stderr)
+                    print ('unknown error')
```

Comment:
Do we actually need this? asv should give us the tools to run benchmarks this seems a bit unnecessary.

Reply:
Let me keep this to run benchmark casually and get a csv format.